src/Text/Trifecta/Parser.hs

```diff
@@ -129,14 +130,23 @@ instance Semigroup a => Semigroup (Parser a) where
   (<>) = liftA2 (<>)
   {-# INLINE (<>) #-}
 
+{- In order to silence '-Wnoncanonical-monoid-instances' we'd need to
+   change the type-signature to emulate the not-yet existing superclass
+   relationship:
+
+instance (Semigroup a, Monoid a) => Monoid (Parser a) where
+  mappend = (<>)
+  mempty = pure mempty
+-}
 instance Monoid a => Monoid (Parser a) where
```

Comment:
I think that would be an acceptable solution. Moreover, we're going to need to support pre-SMP compilers anyway, so we'll have to use `(Semigroup a, Monoid a)` for them anyways. I'm thinking:

```haskell
instance ( Monoid a
#if PRE_SMP_GHC
         , Semigroup a
#endif
         ) => Monoid (Parser a) where ...
```

Problem: I don't know what GHC version `PRE_SMP_GHC` should use.